### PR TITLE
drivers: neural_net: remove obsolete driver

### PR DESCRIPTION
**Pull request by nashif**
This driver is not being used by anything and is not being built or
tested, so remove it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Comment by kartben** [comment]
Frankly I would look at removing GNA altogether at this point. Still experimental after many years, and does not meet the basic requirements of having at least one implementation + code sample.

File: drivers/neural_net/intel_gna.c [hunk 1]
```diff
@@ -1,526 +0,0 @@
...
- */
-
-/** @file
- * @brief Intel GNA device driver
- *
- * Device driver implementation for Intel's
- * Gaussian Mixture Model and Neural Network Accelerator (GNA)
- */
-
-#define DT_DRV_COMPAT intel_gna
```
Comment (kartben) [inline drivers/neural_net/intel_gna.c:16]: Need to also git rm dts/bindings/neural_net/intel,gna.yaml

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
> Frankly I would look at removing GNA altogether at this point. Still experimental after many years, and does not meet the basic requirements of having at least one implementation + code sample.

yes, maybe, but as a second step, I am just applying the general rule that someone, somewhere might be using this, so need to figure removal of APIs, even experimental ones

**Comment by kartben** [comment]
> > Frankly I would look at removing GNA altogether at this point. Still experimental after many years, and does not meet the basic requirements of having at least one implementation + code sample.
> 
> yes, maybe, but as a second step, I am just applying the general rule that someone, somewhere might be using this, so need to figure removal of APIs, even experimental ones

I think we are making it pretty clear that experimental ((in fact, even _unstable_) API might be removed at any time and I really don't see a reason to keep an API that 100% falls in the category of "experimental stuff that unfortunately didn't make it to the next stage". I think we should really try to stay on top of things in housekeeping situations like this one.

FWIW, I don't know much about this GNA stuff, but I am wondering if it would ever be able to move out of Experimental stage anyways, as it seems to be Intel specific (at least the gna.h header says "Public API header file for Intel GNA driver")? So it's not clear how we'd have "[...] at least two implementations on different hardware platforms.".

Thanks!

**Comment by fabiobaltieri** [comment]
Would there be any difference removing it now or later? If there's any driver out there that uses it they'll probably only find out once the apis are gone anyway.

**Comment by nashif** [comment]
> Would there be any difference removing it now or later? If there's any driver out there that uses it they'll probably only find out once the apis are gone anyway.

yes, later is no later than 3.5 rc1. I am not calling for deprecation or anything, this is just about doing it gracefully. The API change will follow while I am checking. I know for a fact there are users of this, so it does not make sense to drop an API to introduce again later.

**Review by fabiobaltieri — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**